File: backend/mr-db-sql/src/main/java/io/featurehub/db/services/SetupSqlApi.java
package io.featurehub.db.services;

import io.ebean.Database;
import io.ebean.annotation.Transactional;
import io.featurehub.db.api.SetupApi;
import io.featurehub.db.model.DbPerson;
import io.featurehub.mr.model.SetupSiteAdmin;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;

@Singleton
public class SetupSqlApi implements SetupApi {
  private final Database database;
  private final Conversions convertUtils;

  @Inject
  public SetupSqlApi(Database database, Conversions convertUtils) {
    this.database = database;
    this.convertUtils = convertUtils;
  }

  @Override
  public boolean initialized() {
    return convertUtils.dbOrganization() != null;
  }

  @Override
  @Transactional
  public boolean setup(SetupSiteAdmin setupSiteAdmin) {
    assert setupSiteAdmin.getEmailAddress() != null && setupSiteAdmin.getName() != null;

    if (!initialized()) {
      database.save(new DbPerson.Builder().email(setupSiteAdmin.getEmailAddress()).name(setupSiteAdmin.getName()));
      return true;
    }
    return false;
  }
}

File: backend/mr-db-sql/src/main/kotlin/io/featurehub/db/services/SetupSqlApi.kt
package io.featurehub.db.services

import io.ebean.Database
import io.ebean.annotation.Transactional
import io.featurehub.db.api.SetupApi
import io.featurehub.db.model.DbPerson
import io.featurehub.mr.model.SetupSiteAdmin
import jakarta.inject.Inject
import jakarta.inject.Singleton

@Singleton
class SetupSqlApi @Inject constructor(private val convertUtils: Conversions) :
  SetupApi {
  override fun initialized(): Boolean {
    return convertUtils.hasOrganisation()
  }

  @Transactional
  override fun setup(setupSiteAdmin: SetupSiteAdmin): Boolean {
    assert(setupSiteAdmin.emailAddress != null && setupSiteAdmin.name != null)
    if (!initialized()) {
      DbPerson.Builder().email(setupSiteAdmin.emailAddress).name(setupSiteAdmin.name).build().save()
      return true
    }
    return false
  }
}
